src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BasisOfPreparationControllerTests.java

```diff
@@ -0,0 +1,171 @@
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.model;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.view;
+
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.servlet.view.UrlBasedViewResolver;
+import uk.gov.companieshouse.web.accounts.controller.smallfull.BasisOfPreparationController;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.smallfull.notes.accountingpolicies.BasisOfPreparation;
+import uk.gov.companieshouse.web.accounts.service.smallfull.BasisOfPreparationService;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class BasisOfPreparationControllerTests {
+
+    private MockMvc mockMvc;
+
+    @Mock
+    private List<ValidationError> validationErrors;
+
+    @Mock
+    private BasisOfPreparationService basisOfPreparationService;
+
+    @InjectMocks
+    private BasisOfPreparationController controller;
+
+    private static final String COMPANY_NUMBER = "companyNumber";
+
+    private static final String TRANSACTION_ID = "transactionId";
+
+    private static final String COMPANY_ACCOUNTS_ID = "companyAccountsId";
+
+    private static final String BASIS_OF_PREPARATION_PATH = "/company/" + COMPANY_NUMBER +
+                                                            "/transaction/" + TRANSACTION_ID +
+                                                            "/company-accounts/" + COMPANY_ACCOUNTS_ID +
+                                                            "/small-full/basis-of-preparation";
+
+    private static final String REVIEW_PATH = "/company/" + COMPANY_NUMBER +
+                                                "/transaction/" + TRANSACTION_ID +
+                                                "/company-accounts/" + COMPANY_ACCOUNTS_ID +
+                                                "/small-full/review";
+
```

Comment:
we could create a constant for SMALL_FULL_PATH 

```
private static final String SMALL_FULL_PATH = "/company/" + COMPANY_NUMBER +
        "/transaction/" + TRANSACTION_ID +
        "/company-accounts/" + COMPANY_ACCOUNTS_ID +
        "/small-full";

private static final String BASIS_OF_PREPARATION_PATH = SMALL_FULL_PATH + "/basis-of-preparation"
private static final String REVIEW_PATH  =  SMALL_FULL_PATH + "/review"
```

Reply:
Added 👍